src/pip/_internal/self_outdated_check.py

```diff
@@ -113,10 +109,8 @@ def was_installed_by_pip(pkg):
     This is used not to display the upgrade message when pip is in fact
     installed by system package manager, such as dnf on Fedora.
     """
-    dist = get_distribution(pkg)
-    if not dist:
-        return False
-    return "pip" == get_installer(dist)
+    dist = get_environment().get_installed_distribution(pkg)
```

Comment:
Is there a reason we are not calling `get_distribution` directly here?

Reply:
Not really. This one was changed before the rebase, as an example how the migration should be done, and I didn’t think too much when I rebased. Indeed the patch would have read batter if I did not include this change, I’ll change it (back).